core/src/main/java/org/radargun/stages/test/Stressor.java

```diff
@@ -157,13 +158,15 @@ private void runInternal() {
    }
 
    public <T> T makeRequest(Invocation<T> invocation, boolean countForTx) throws OperationLogic.RequestException {
+
+      if (recording()) {
+         requests = stats.requestSet();
```

Comment:
The requestSet should be created only when we're about to start a transaction - when there's ongoing transaction with multiple writes it shouldn't do it on each request.

Reply:
Removed the commit